### PR TITLE
Updates TGUI from V3 To V4.3 Hotfix 1 

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request

Updates tgui from TGUI 3 to TGUI 4.3, adds lots of new stuff and asset handling and new stuff to be used on UI's
such as tgui_alert and tgui_input_list which should replace lots of input() 

New things that are in most notably: 
TGUI windows now use pooling, handled clientside i believe and makes opening of subsequent UI's faster and less resourcefull.

Comes with support for tgchat and tg stat panel,  probably port these in the future, most framework is basically there, except in DM where you'll need to code stuff.

UI_Interact now has been reduced alot, check changes.

UI's that previously used ingame icons by changing basehtml(None did in here) now have a proper way and use ui_asset() to send asset packages.

<Fragment></Fragment> Fragments without keys don't need to be imported anymore and can be used as <></>, Fragments with key stay the same.

Adds a stack component which is basically flex but with auto spacing and etcetera.

Adds tgui_alert(), fancy thing that replaces tgalert and inputs:
image

and adds
tgui_input_list() that replaces:
image

with:

image

## Why It's Good For The Game
TGui UPDATE YOOO

## Changelog
:cl:
code: Updates from TGUI3 to TGUI 4.3.
code: adds tgui_alert and tgui_input_list, the drone control uses tgui_alert.
/:cl:

### CODE REVIEW

**Comment by neohenry** [comment]
The prophesied son is here.

**Comment by SplinterGP** [comment]
now to look at all files changed and se where i fucked up

File: code/datums/wires/_wires.dm [hunk 1]
```diff
@@ -242,11 +242,13 @@
 		return ..()
 	return UI_CLOSE
 
-/datum/wires/ui_interact(mob/user, ui_key = "wires", datum/tgui/ui = null, force_open = FALSE, \
-							datum/tgui/master_ui = null, datum/ui_state/state = GLOB.physical_state)
-	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+/datum/wires/ui_state(mob/user)
+	return GLOB.physical_state
+
+/datum/wires/ui_interact(mob/user, datum/tgui/ui,)
```
Comment (SplinterGP) [inline code/datums/wires/_wires.dm:248]:
```suggestion
/datum/wires/ui_interact(mob/user, datum/tgui/ui)
```

File: code/game/objects/machinery/vending/vending.dm [hunk 2]
```diff
@@ -389,12 +386,11 @@ GLOBAL_LIST_INIT(vending_white_items, typecacheof(list(
...
 
-/obj/machinery/vending/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, \
-										datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
-	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/vending/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
 
 	if(!ui)
-		ui = new(user, src, ui_key, "Vending", name, ui_x, ui_y, master_ui, state)
+		ui = new(user, src, "Vending")
```
Comment (SplinterGP) [inline code/game/objects/machinery/vending/vending.dm:393]: Name

File: code/controllers/subsystem/shuttle.dm [hunk 3]
```diff
@@ -487,11 +487,13 @@ SUBSYSTEM_DEF(shuttle)
 		preview_shuttle.jumpToNullSpace()
 	preview_shuttle = null
 
+/datum/controller/subsystem/shuttle/ui_state(mob/user)
+	return GLOB.admin_state
 
-/datum/controller/subsystem/shuttle/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.admin_state)
-	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+/datum/controller/subsystem/shuttle/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src ,ui)
```
Comment (SplinterGP) [inline code/controllers/subsystem/shuttle.dm:494]:
```suggestion
	ui = SStgui.try_update_ui(user, src, ui)
```

**Comment by SplinterGP** [comment]
Almost there, need to fix a few bugs first with UI's first, Vending UI, Orbit UI(i broke this myself to fix a bug, fix soon), and CameraConsole (tablets and other camera on UI thing are breaking).

**Comment by SplinterGP** [comment]
tibi said to TM

**Comment by BraveMole** [comment]
ALERT SELF TM. SPLINTER IS GOING ROGUE

**Comment by SplinterGP** [comment]
> ALERT SELF TM. SPLINTER IS GOING ROGUE

i mean i've self tm'd before, suffer

**Review by TiviPlus — APPROVED** [review]
Github is lagging

**Comment by SplinterGP** [comment]
wait bot removes tmc ? pog ?

**Comment by SplinterGP** [comment]
> Github is lagging

rip

**Comment by TiviPlus** [comment]
@SplinterGP nob want fite

**Comment by SplinterGP** [comment]
> @SplinterGP noob

no u

**Pull request merged**